staging/src/k8s.io/client-go/dynamic/scheme.go

```diff
@@ -77,6 +79,8 @@ func (s basicNegotiatedSerializer) SupportedMediaTypes() []runtime.SerializerInf
 	return []runtime.SerializerInfo{
 		{
 			MediaType:        "application/json",
```

Comment:
is constructing these a normal thing to do? is the requirement to populate these two additional fields going to catch consumers off guard?

Reply:
There were only four spots in the code where these are constructed - codecfactory, two tests, and the dynamic watch cache.

Reply:
I wasn't sure if out-of-tree callers had reason to construct these normally. in their absence, what happens?

Reply:
You wouldn't negotiate on the server side.  Negotiate is server only today.  I can flag an error in negotiate if they are unset.

Reply:
Flagged an error if unset when initializing endpoint handlers (currently the only code that calls negotiate).